### PR TITLE
Bugfix for pasting on macOS

### PR DESCRIPTION
**Pull request by xukai92**
This fixes https://github.com/tecosaur/emacs-everywhere/issues/19 on my local.

How did I find the fix?
- Copying from the selection is calling `CMD-c` directly (https://github.com/tecosaur/emacs-everywhere/blob/master/emacs-everywhere.el#L387), so I wonder why not pasting.
- `emacs-anywhere`'s macOS script is calling `CMD-c` as well (https://github.com/zachcurry/emacs-anywhere/blob/master/bin/osx#L55)

### CODE REVIEW

**Comment by d1egoaz** [comment]
@xukai92 not related to this PR, how do you use this in MacOS?

I'm currently using an Alfred workflow for this, but unsure how other people uses it

**Comment by d1egoaz** [comment]
> @xukai92 not related to this PR, how do you use this in MacOS?
> 
> I'm currently using an Alfred workflow for this, but unsure how other people uses it

I saw some usages here https://github.com/tecosaur/emacs-everywhere/issues/1

**Comment by xukai92** [comment]
I'm using the built-in app Automator to [create a Service](https://developer.apple.com/library/archive/documentation/LanguagesUtilities/Conceptual/MacAutomationScriptingGuide/MakeaSystem-WideService.html) to run `doom everywhere` (whatever the command you use to trigger emacs-everywhere) and register a shorcut using macOS's built-in settings at `System Preferences > Keyboard > Shortcuts > Services`.

**Comment by d1egoaz** [comment]
> I'm using the built-in app Automator to [create a Service](https://developer.apple.com/library/archive/documentation/LanguagesUtilities/Conceptual/MacAutomationScriptingGuide/MakeaSystem-WideService.html) to run `doom everywhere` (whatever the command you use to trigger emacs-everywhere) and register a shorcut using macOS's built-in settings at `System Preferences > Keyboard > Shortcuts > Services`.

Thanks!

I tried this, but it doesn't paste the result 🤷 
image

BTW in MacOS BigSur it's called a Quick Action instead of a Service.

**Comment by xukai92** [comment]
This is what mine looks like:

image

Maybe try to match some of the settings of mine to see if it works?

**Comment by xukai92** [comment]
This works for me at least. Would it be a Doom Emacs thing or so? I don&rsquo;t know.

**Comment by d1egoaz** [comment]
Thanks @xukai92 

I think I found the issue, I needed to allow `bash` to control my computer in Security & Privacy, Privacy, Accessibility

image

BTW, paste does work for me, and it looks like internet says that using `(the clipboard as text)` to paste is fine

**Comment by d1egoaz** [comment]
I couldn't reproduce https://github.com/tecosaur/emacs-everywhere/issues/19 on my machine, it works fine

**Comment by d1egoaz** [comment]
However, `(the clipboard as text)` breaks emojis https://github.com/tecosaur/emacs-everywhere/issues/12

It's fixed by using Command-V, what this PR is doing.

**Comment by tecosaur** [comment]
Interesting, I'm getting the distinct impression that Cmd-V may be better then, and it's good to get some Mac troubleshooting/setup info. I think it may be a good idea to add a bit to the readme on that at some point.

**Comment by d1egoaz** [comment]
An image to help MacOS users to see where they can add the global shortcut

**Comment by HyunggyuJang** [comment]
I can confirm that with this commit, I could solve the problem with <span class="underline">macOS</span>.

Actually today, consuming my whole time, I tried to make emacs-everywhere to work, attracted by henrik, who, as you know *the* author of doom emacs, and used this application in his qutebrowser configuration file. Being fascinated by this setting, today, I decided to work it out; however, betraying my initial expectation, it hardly worked.

And I found this conversation, and lo and behold, it starts working!

To sum up, this pull request definitely be merged to the main branch, especially helpful for international language users. I actually encountered some weird behavior with the current `applescript` command, `(clipboard as text)`, when I typed it manually in terminal, the text stored in clipboard spat out in Japanese words, even if I set the keyboard layer as US. The thing is, my mac has JIS layout by default.

Thanks for bringing up this pull request!!

**Comment by tecosaur** [comment]
With these comments, I am now convinced that this is with merging ... at least until the next bug report :stuck_out_tongue:.

**Comment by Bounga** [comment]
Hi! I followed all the advices I found here but with no luck.

I created a Mac OS "quick action" which calls the `doom everywhere` script. It works, the keybinding creates a new Emacs frame popup in which I can type my text. Then when I `C-c C-c`, `C-q-f` or `C-x 5 0` it brings me back to the originating app but without pasting the text. My text is in the clipboard since `cmd-v` makes it appear.

I allowed Emacs and Bash in system accessibility preferences. My Doom is up-to-date.

Any clue?

**Comment by HyunggyuJang** [comment]
Can you check the following commit solves your problem https://github.com/HyunggyuJang/emacs-everywhere/commit/9229fae0ac1b8ebda17bcd136a1f2bdbe0d8da24

Relevant discussion can be found at #24

**Comment by Bounga** [comment]
> @Bounga Can you check the following commit solves your problem [HyunggyuJang@9229fae](https://github.com/HyunggyuJang/emacs-everywhere/commit/9229fae0ac1b8ebda17bcd136a1f2bdbe0d8da24)
> 
> Relevant discussion can be found at #24

I’ll try using this commit as soon as I figure out how to do it in Doom 😄. I’ll let you know if it fixes my issue.

**Comment by Bounga** [comment]
> @Bounga Can you check the following commit solves your problem [HyunggyuJang@9229fae](https://github.com/HyunggyuJang/emacs-everywhere/commit/9229fae0ac1b8ebda17bcd136a1f2bdbe0d8da24)
> 
> Relevant discussion can be found at #24

So now that latest version of emacs-everywhere was merged into Doom I had a chance to try it. It doesn’t solve the issue for me. No automatic paste, I still have to do cmd-c for the content of the clipboard to be pasted into the targeted text area

**Comment by tecosaur** [comment]
Gah! I had my hopes up with that PR. Sorry to hear it hasn't fixed things for you, and thanks for letting me know.

Hopefully other Mac users might be able to help try to work out another fix+PR for this.

**Comment by dianrob** [comment]
For me I had to all AEServer in Security & Privacy -> Privacy -> Accessibility to make the pasting work in FF.

Yet it doesnt work in qutebrowser at all. Anybody have ideas? :<

**Comment by yevhenshymotiuk** [comment]
It seems that you have to (at least) allow it for every specific application you are going to use it in
